cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -106,7 +106,7 @@ set_compiler_property(PROPERTY cstd -std=)
 
 if (NOT CONFIG_NEWLIB_LIBC AND
     NOT COMPILER STREQUAL "xcc" AND
-    NOT ZEPHYR_TOOLCHAIN_VARIANT STREQUAL "espressif" AND
+    NOT CONFIG_SOC_ESPRESSIF AND
```

Comment:
@stephanosio We have discussed about this before. The necessary work to get `hal_espressif` cleaned up regarding `nostdinc` is complex. I ask you to accept this change as is for now.

Reply:
This is a very ugly hack that effectively allows code to make use of the newlib headers without linking the newlib library, which may or may not work depending on what you are using.

I am ok with this as a temporary workaround; but, as noted in the commit message, this needs to be fixed at some point by either removing the stdinc dependencies or adding the required stdinc dependency implementations in `hal_espressif`.

Reply:
Sure, in fact that hack was there for a long time. I explicit added the workaround in commit message in order to tell that this will be re-worked as you noted.